### PR TITLE
Added Language file for Norwegian keyboard layout

### PR DESCRIPTION
**Pull request by erikbra**

### CODE REVIEW

File: src/Layouts/NO_QWERTY.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    'shift+vk_oem_7' : 'Æ',
+    'shift+vk_oem_102' : '>',
+
+    'vk_oem_minus'  : '+',
+    'vk_oem_plus'   : '\\',
+    'vk_oem_comma'  : ',',
+    'vk_oem_period' : '.',
+
+    'shift+vk_oem_minus'  : '?',
+    'shift+vk_oem_plus'   : '+',
```
Comment (aioutecism) [inline src/Layouts/NO_QWERTY.ts]: Should this be ` ?

**Comment by aioutecism** [comment]
Ref

**Comment by aioutecism** [comment]
Ref
keyboard_layout_norwegian

File: src/Layouts/NO_QWERTY.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import {LayoutMap} from './Layout';
+
+export const id = 'NO QWERTY';
```
Comment (aioutecism) [inline src/Layouts/NO_QWERTY.ts]: Should we make the id easier to understand by using `Norwegian QWERTY`?

**Comment by aioutecism** [comment]
Cloud you add this layout option to `README.md`?

File: src/Layouts/NO_QWERTY.ts (hunk 2 above)
Comment (erikbra) [inline src/Layouts/NO_QWERTY.ts]: Weeelll.. NO is the ISO3166 two-letter country code for Norway, so it is actually a sensible choice. The problem is that it's also the word 'NO'. Say you get an Azerbaijanian layout, Would you call it 'AZ' or Azerbaijanian? Maybe we should make one more example, just to clarify that this is actually the country code, and not "Not a QWERTY keyboard mapping"?

**Comment by erikbra** [comment]
BTW, I couldn't get testing the layout to work. I opened VSCode, but then you said to "Open the root folder" and press F5. Where do I 'install' the plugin from source?

File: src/Layouts/NO_QWERTY.ts (hunk 1 above)
Comment (erikbra) [inline src/Layouts/NO_QWERTY.ts]: Which one do you mean?

Comment (erikbra) [inline src/Layouts/NO_QWERTY.ts]: Yes, you are correct. But, that one's also a dead key. Have commented it out, and commented that it's a dead key.

File: src/Layouts/NO_QWERTY.ts (hunk 2 above)
Comment (aioutecism) [inline src/Layouts/NO_QWERTY.ts]: I'm not familiar with ISO3166 code ;)
This is great. I think a comment above the `export const id` line would be enough clear for contributors.
```
# ISO3166 country code: https://www.iso.org/obp/ui/#search/code/
export const id = 'NO QWERTY';
```

**Comment by aioutecism** [comment]
For testing, you should

1. Open your command line application (PowerLine or any of your favorites).
1. `cd path/to/your/git/clone`.
1. Run `npm install` (You must have node installed).
1. Open VSCode.
1. "File / Open.." and open `path/to/your/git/clone`.
1. Press `F5`.

Let me know where you stuck in.

**Comment by aioutecism** [comment]
You may find [this](https://code.visualstudio.com/docs/extensions/debugging-extensions) helpful.

File: src/Layouts/NO_QWERTY.ts (hunk 1 above)
Comment (aioutecism) [inline src/Layouts/NO_QWERTY.ts]: I see.

**Comment by erikbra** [comment]
Sorry, I forgot to "Open folder", so I couldn't open it. Now I managed, but it basically just lets me enter ONE character in the new dev-vscode window that shows up, no matter which keyboard layout I use.

**Comment by aioutecism** [comment]
I tried your branch, and realized I forgot some code you need to add:

```diff
diff --git a/src/Layouts/Layout.ts b/src/Layouts/Layout.ts
index 8b10e4f..7d22031 100644
--- a/src/Layouts/Layout.ts
+++ b/src/Layouts/Layout.ts
@@ -1,4 +1,5 @@
 import * as US_QWERTY from './US_QWERTY';
+import * as NO_QWERTY from './NO_QWERTY';

 export interface LayoutMap {
     [key: string]: string;
@@ -52,3 +53,4 @@ export class Layout {
 }

 Layout.register(US_QWERTY.id, US_QWERTY.map);
+Layout.register(NO_QWERTY.id, NO_QWERTY.map);
```

**Comment by erikbra** [comment]
I just don't get it. I cannot enter more than ONE character after entering insert mode. Then nothing happens. Even with US keyboard layout (same with Norwegian). Don't think not mapping the dead keys work either. Get error on load when I add your edits, and have commented out the dead keys.

**Comment by magbeat** [comment]
@aioutecism I have the same problem as @erikbra. I have added Swiss German layout in #48

**Comment by aioutecism** [comment]
@erikbra 
I'm not sure what your problem is.
Would you mind I merge this p-r and publish a new version for you to test?

**Comment by aioutecism** [comment]
@magbeat 
Thanks! I'll check it now.

**Comment by aioutecism** [comment]
Hi @erikbra 

I merged the keyboard layout main branch into master.
Cloud you solve the conflicts and push again?

**Comment by aioutecism** [comment]
Since https://github.com/Microsoft/vscode/issues/1432 is addressed.
All layouts should work fine after we update the extension to use the new APIs VSCode provide.
I'm closing this p-r.

If you'd like to track status of obtaining new APIs, please subscribe to [this](#50).

**Pull request closed without merging**